test/shared_utilities/variable_types.jl

```diff
@@ -17,97 +17,99 @@ using ClimaLSM.Domains: HybridBox, Column, Point
 using ClimaLSM.Domains: coordinates
 using ClimaLSM.Canopy: AbstractCanopyComponent
 
-@testset "Default model" begin
-    struct DefaultModel{FT} <: AbstractModel{FT} end
-    dm = DefaultModel{Float32}()
-    @test ClimaLSM.prognostic_vars(dm) == ()
-    @test ClimaLSM.prognostic_types(dm) == ()
-    @test ClimaLSM.prognostic_domain_names(dm) == ()
-    @test ClimaLSM.auxiliary_vars(dm) == ()
-    @test ClimaLSM.auxiliary_types(dm) == ()
-    @test ClimaLSM.auxiliary_domain_names(dm) == ()
-
-    x = [0, 1, 2, 3]
-    dm_exp_tendency! = make_exp_tendency(dm)
-    @test dm_exp_tendency!(x[1], x[2], x[3], x[4]) == nothing
-    @test x == [0, 1, 2, 3]
-
-    dm_compute_exp_tendency! = make_compute_exp_tendency(dm)
-    @test dm_compute_exp_tendency!(x[1], x[2], x[3], x[4]) == nothing
-    @test x == [0, 1, 2, 3]
-
-    dm_imp_tendency! = make_imp_tendency(dm)
-    @test dm_imp_tendency!(x[1], x[2], x[3], x[4]) == nothing
-    @test x == [0, 1, 2, 3]
-
-    dm_compute_imp_tendency! = make_compute_imp_tendency(dm)
-    @test dm_compute_imp_tendency!(x[1], x[2], x[3], x[4]) == nothing
-    @test x == [0, 1, 2, 3]
-
-    dm_update_aux! = make_update_aux(dm)
-    @test dm_update_aux!(x[1], x[2], x[3]) == nothing
-    @test x == [0, 1, 2, 3]
-end
+for FT in (Float32, Float64)
```

Comment:
Im not sure we need the different values of FT for this test? (I know it doesnt really make a difference and is good practice. but not sure if we want to be a little more judicious about where we test multiple types, since doing everywhere will double CI time (I think?))

Reply:
I agree, I don't think we need both FTs for this test. On one hand, it might be good practice to cover both FTs everywhere, and I don't think it will quite double CI time since precompilation still just happens once. I'm okay with either covering both FTs in all tests, or doing it for just the ones we think require it (maybe in the latter case we could add a note justifying why we don't need to cover both FTs)

Reply:
I think my preference would be to only do where required but it's your call! youre right that it wont double CI.